math/geometry.h

```diff
@@ -313,7 +342,7 @@ bool RayIntersect(const Eigen::Matrix<T, 2, 1>& ray_source,
   T line_intersect_param = line_to_source.dot(perpendicular) / denomenator;
 
   if (ray_intersect_param >= 0 && line_intersect_param >= 0 &&
-    line_intersect_param <= 1) {
+      line_intersect_param <= 1) {
```

Comment:
Comparisons should preserve types: `T(0)` , `T(1)`, etc. instead of `0`, `1`, etc.

Reply:
Could you also please fix the other older instances of this?

Reply:
Done.